test/LinkedListTest.cpp

```diff
@@ -39,13 +39,28 @@ TEST(LinkedListTest, MaxNumber){
     validateLinkedList(head, INT_MAX);
 }
 
+TEST(LinkedListTest, CleanMemory){
+    const int SIZE = 6;
+    Node* head = createList(SIZE);
+    validateLinkedList(head, SIZE);
+    cleanMemory(head);
+    ASSERT_NE(nullptr, head);
+}
+
 void validateLinkedList(Node* head, int size){
-    for(int i = 0; i < size; i++ ){
-        int currentData = head->data;
-        printf("Checking if %d is equals to %d \n", currentData, DEFAULT_VALUE);
-        GTEST_ASSERT_EQ(currentData, DEFAULT_VALUE);
+    int count = 0;
+
+    while(head && count <= size){
```

Comment:
Before, we check if the data was correct, and verify that the last node is null. If the size of the list is, for example smaller, an exception will occur as we can not access the data value due a null pointer exception.
If the list is bigger, the assertion will fail as head is not null.

In order to avoid exceptions and have more information, I added a `while`and a counter. So we check when the head until is null or we iterate 'size' times.

We add two assertion for verify that the size is correct and is tail node (next node is null)